File: SPRING/Modern_Hospital_Management_System/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/DietHistoryService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.entity.DietHistory;
import java.util.List;

public interface DietHistoryService {
    DietHistory create(DietHistory history);
    List<DietHistory> getAll();
    List<DietHistory> getByPatientId(Long patientId);
    List<DietHistory> getByDietAssignmentId(Long assignmentId);
}

File: SPRING/Hospital_Management_system_angular/src/app/services/dietary/diet-history.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { DietHistory } from '../../models/dietary/diet-history.model';

@Injectable({ providedIn: 'root' })
export class DietHistoryService {
  private api = environment.apiUrl + 'diet-history';

  constructor(private http: HttpClient) {}

  getAll(): Observable<DietHistory[]> {
    return this.http.get<DietHistory[]>(this.api);
  }
  getById(id: number): Observable<DietHistory> {
    return this.http.get<DietHistory>(`${this.api}/${id}`);
  }
  getByPatientId(patientId: number): Observable<DietHistory[]> {
    return this.http.get<DietHistory[]>(`${this.api}/patient/${patientId}`);
  }
  getByDietAssignmentId(assignmentId: number): Observable<DietHistory[]> {
    return this.http.get<DietHistory[]>(`${this.api}/assignment/${assignmentId}`);
  }
  create(data: DietHistory): Observable<DietHistory> {
    return this.http.post<DietHistory>(this.api, data);
  }
}
